pkg/alertmanager/alertmanager.go

```diff
@@ -72,7 +72,15 @@ type Alertmanager struct {
 	active    bool
 }
 
-var webReload = make(chan chan error)
+var (
+	webReload = make(chan chan error)
+
+	// Workaround a bug in the alertmanager which doesn't register the
+	// metrics in the input registry but to the global default one.
+	// TODO change once the vendored alertmanager will have this PR merged into:
+	//      https://github.com/prometheus/alertmanager/pull/2200
+	dispatcherMetrics = dispatch.NewDispatcherMetrics(prometheus.NewRegistry())
```

Comment:
I don't get how this is a workaround for that bug? `dispatch.NewDispatcherMetrics` still registers into default registry.

Reply:
Yes, but we need to pass something otherwise it panics.

Reply:
Makes sense now, thanks!